src/hotspot/share/gc/shenandoah/shenandoahHeap.cpp

```diff
@@ -3185,13 +3186,13 @@ void ShenandoahHeap::verify_rem_set_at_mark() {
             // else, object's start is marked dirty and obj is not an objArray, so any interesting pointers are covered
             if (!scanner->verify_registration(obj_addr, ctx)) {
               ShenandoahAsserts::print_failure(ShenandoahAsserts::_safe_all, obj, obj_addr, NULL,
-                                            "Verify init-mark remembered set violation", "object not properly registered", __FILE__, __LINE__);
+                                               "Verify init-mark remembered set violation", "object not properly registered", __FILE__, __LINE__);
             }
             obj_addr += obj->size();
           } else {
             // This object is not live so we don't verify dirty cards contained therein
-            assert(ctx->top_at_mark_start(r) == top, "Expect tams == top at start of mark.");
-            obj_addr = ctx->get_next_marked_addr(obj_addr, top);
+            assert(tams != nullptr, "If object is not live, ctx and tams should be non-null");
```

Comment:
Might need to rebase these changes after integrating #197 .

Reply:
So far, git seems to feel like there are "no conflicts".  This code is identical to what I delivered in PR197.